### PR TITLE
added config normalization

### PR DESCRIPTION
**Pull request by sjka**
Signed-off-by: Simon Kaufmann <simon.kfm@googlemail.com>

### CODE REVIEW

**Comment by maggu2810** [comment]
Why is an empty string equal to null?

IIRC there has been some comments about the Hue bridge username configuration and we stated that a null one differs from an empty one (have to look at this issue). Regardless the Hue issue, I assume an empty string and null could differ some time (e.g. null means unset and "" means I set an empty one). Or do we never support the difference between empty string and null for configuration values?

**Comment by kaikreuzer** [comment]
I agree with @maggu2810's comment.

**Comment by sjka** [comment]
okay, convinced. I updated the PR. 

(I'm just not sure why the IP validation failed, the sign off is there...?)

**Comment by maggu2810** [comment]
Do you want to use your googlemail or your itemis mail address?

```
commit afc4a1458aa3161dae6b4ae29dcfa60a3a8f82b0
tree c748356d14dd68f225aef057f6889a0d26636155
parent e997fbd181b59dee7434541981f1100ba9a22702
author Simon Kaufmann <simon.kfm@googlemail.com> 1461327897 +0200
committer Simon Kaufmann <simon.kaufmann@itemis.de> 1461831906 +0200
```

**Comment by maggu2810** [comment]
IMHO If you use your googlemail for commit (instead of the itemis one), it should work.

**Comment by sjka** [comment]
No idea why I have missed that... Thanks!

File: bundles/config/org.eclipse.smarthome.config.core/src/main/java/org/eclipse/smarthome/config/core/ConfigUtil.java [hunk 1]
```diff
@@ -43,7 +46,41 @@
...
+    }
+
+    /**
+     * Normalizes the given configuration according to their config description.
+     *
+     * By doing so, it tries to convert types on a best-effort basis. The result will contain
+     * BigDecimals, Strings and Booleans wherever a conversion of similar types was possible.
+     *
+     * However, it does not check for general correctness of types. This can be done using the
+     * ConfigDescriptionCalidator.
```
Comment (maggu2810) [inline bundles/config/org.eclipse.smarthome.config.core/src/main/java/org/eclipse/smarthome/config/core/ConfigUtil.java]: typo: Calidator => Validator

**Comment by maggu2810** [comment]
One typo found, but looks good to me.

**Pull request merged**